src/script/conversation/ConversationRepository.ts

```diff
@@ -1741,19 +1748,25 @@ export class ConversationRepository {
       return localProteusConversation || this.fetchConversationById(proteusConversationId);
     }
 
-    const isWebSocketEvent = source === EventSource.WEBSOCKET;
     const isConnectionAccepted = connectionEntity.isConnected();
 
     // Check what protocol should be used for 1:1 conversation
     const {protocol, isMLSSupportedByTheOtherUser, isProteusSupportedByTheOtherUser} =
       await this.getProtocolFor1to1Conversation(otherUserId);
 
+    const isWebSocketEvent = source === EventSource.WEBSOCKET;
+    const shouldDelayMLSGroupEstablishment = isWebSocketEvent && isMLSSupportedByTheOtherUser;
```

Comment:
No need to delay anything if the other user does not support mls - we will not receive a welcome message in this case anyway.